certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -896,8 +898,50 @@ def test_preferred_chain(context: IntegrationTestsContext) -> None:
 
         dumped = misc.read_certificate(cert_path)
         assert 'Issuer: CN={}'.format(expected) in dumped, \
-               'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
+            'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
 
         with open(conf_path, 'r') as f:
             assert 'preferred_chain = {}'.format(requested) in f.read(), \
-                   'Expected preferred_chain to be set in renewal config'
+                'Expected preferred_chain to be set in renewal config'
+
+
+def test_certificate_validity(context: IntegrationTestsContext) -> None:
+    """Test that --certificate-validity is requesting certificates of the appropirate validity"""
+    if context.acme_server == 'boulder-v2':
+        pytest.skip('Boulder does not support NewOrder notAfter.')
+
+    context.certbot(['certonly', '--cert-name', 'newname', '-d', context.get_domain('newname'),
+                     '--certificate-validity', '86400'])
+
+    want_not_after = datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=1)
+
+    cert_path = join(context.config_dir, 'live', 'newname', 'cert.pem')
+
+    not_after_first = get_not_after(cert_path)
+    tc = unittest.TestCase()
+    tc.assertAlmostEqual(not_after_first, want_not_after, delta=datetime.timedelta(minutes=5))
+
+    context.certbot(['renew', '--force-renewal'])
+
+    not_after_renewal = get_not_after(cert_path)
+    tc = unittest.TestCase()
+    tc.assertAlmostEqual(not_after_renewal, want_not_after, delta=datetime.timedelta(minutes=5))
+
+    assert not_after_first != not_after_renewal
+
+
+def get_not_after(cert_path: str) -> datetime.datetime:
+    """Gets a certificate's not_after field.
+
+    :param str cert_path: Path to a certificate
+    """
+    cert = misc.get_certificate(cert_path)
+    not_after_raw = cert.get_notAfter()
+    not_after_ascii = '1990-01-01T00:00:00+00:00'
+
+    if not_after_raw is not None:
+        not_after_ascii = not_after_raw.decode('ascii')
```

Comment:
What is the logic of having a fallback date here?

If we expect `not_after_raw` to always have value, we could do something like `assert not_after_raw is not None` and let the test crash.